cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Node.kt

```diff
@@ -164,13 +165,25 @@ open class Node : IVisitable<Node>, Persistable, LanguageProvider, ScopeProvider
             this.nextEOGEdges = PropertyEdge.transformIntoOutgoingPropertyEdgeList(value, this)
         }
 
+    /** Incoming data flow edges */
     @Relationship(value = "DFG", direction = Relationship.Direction.INCOMING)
     @PopulatedByPass(DFGPass::class, ControlFlowSensitiveDFGPass::class)
-    var prevDFG: MutableSet<Node> = HashSet()
+    var prevDFGEdges: MutableList<PropertyEdge<Node>> = mutableListOf()
+        internal set
 
+    /** Virtual property for accessing [prevDFGEdges] without property edges. */
     @PopulatedByPass(DFGPass::class, ControlFlowSensitiveDFGPass::class)
-    @Relationship(value = "DFG")
-    var nextDFG: MutableSet<Node> = HashSet()
+    var prevDFG: MutableSet<Node> by PropertyEdgeSetDelegate(Node::prevDFGEdges, false)
```

Comment:
Do we really want to keep this as a `Set`, when the internal representation is a list anyway? Then we would not need the additional delegate.

Reply:
This actually makes me wonder why the other fields are `List`s and we allow duplicate entries (i.e., we don't filter that in the `add...` methods as far as I remember).

Reply:
Good point. Probably all of them should be a set instead of a list?

Reply:
Probably. But this could lead to a bigger changes and might even be API breaking for users

Reply:
We will keep the `Set` here and later review other edges